### ISSUE TITLE
Specific price is aplying to the wrong combination

### ISSUE THREAD

**Issue by isaacrj** [PrestaShop/PrestaShop#20949]
**Prestashop version**
1.7.6.7

**Describe the bug**
When editing a product with several combinations and defining a specific price for one of those combinations, the specific price is saved as applying to the next combination on the list, not to the one being edited.

For example, we have a book that is sold in two different sizes: big and small. If I define a specific price for the big size combination and save it, it appears as applied to the small size combination and vice versa.

We have tried to disable any non native Prestashop addons, deleting the product and recreating it with again with all of its combinations, but the issue persists.

**Expected behavior**

When adding a specific price for a combination, it is expected for this price to apply to that combination.

**Steps to reproduce**

Edit a product with several combinations.  Define a specific price for one of those combinations. When saved, the list of specificprices shows the discount as applied to the next combination in the list, instead of applying it to the combination being edited.

**Screenshots**

![01 -Combination modal](https://user-images.githubusercontent.com/13519885/92874173-12c46d00-f408-11ea-8dce-d16cb151a392.png)
![02 - Specific price result](https://user-images.githubusercontent.com/13519885/92874177-135d0380-f408-11ea-8d3c-f4b39ec844b2.png)

**Comment by florine2623** [comment]
Hi @isaacrj ,
I can't reproduce your issue on PS 1.7.6.7.
In my case I apply the specific price on the white product.
![Capture d’écran 2020-09-11 à 11 47 57](https://user-images.githubusercontent.com/16019289/92906554-da7f5780-f424-11ea-8102-4fa9fb8f0ee1.png)
![Capture d’écran 2020-09-11 à 11 48 03](https://user-images.githubusercontent.com/16019289/92906560-dc491b00-f424-11ea-8206-31a56a51c011.png)
![Capture d’écran 2020-09-11 à 11 48 48](https://user-images.githubusercontent.com/16019289/92906568-deab7500-f424-11ea-8ba6-54c081e0596e.png)

I suggest you to contact the support team to get help on that issue as it seems to be a specific issue to your shop.
https://addons.prestashop.com/en/388-support

Thanks!

**Comment by isaacrj** [comment]
Hi again, we have looked into this issue and we have run tests in all of the PrestaShop installations that we manage.
We have been able to reproduce the bug in some of them, while others just work ok.  

We have found that the bug manifests itself on those installations where running the query 
```
SELECT * FROM `ps_product_attribute` WHERE id_product = <id_product>
```
returns results that are NOT ordered by its primary key.

Is it possible that there is an "ORDER BY id_product_attribute" clause missing somewhere?
